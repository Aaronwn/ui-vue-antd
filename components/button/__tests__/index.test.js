import Button from '../index'
import Icon from '../../icon'
import { mount } from '@vue/test-utils'
import { renderToString } from '@vue/server-test-utils'
import Vue from 'vue'

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = renderToString({
      render () {
        return <Button>Follow</Button>
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('create primary button', () => {
    const wrapper = mount({
      render (h) {
        return <Button type='primary'>按钮</Button>
      },
    })
    expect(wrapper.contains('.ant-btn-primary')).toBe(true)
  })

  it('renders Chinese characters correctly', (done) => {
    const wrapper = mount(
      {
        render (h) {
          return <Button>按钮</Button>
        },
      }
    )
    expect(wrapper.text()).toBe('按 钮')

    const wrapper1 = renderToString(
      {
        render (h) {
          return <Button icon='search'>按钮</Button>
        },
      }
    )
    expect(wrapper1).toMatchSnapshot()

    const wrapper2 = renderToString(
      {
        render (h) {
          return <Button><Icon type='search' />按钮</Button>
        },
      }
    )
    expect(wrapper2).toMatchSnapshot()

    const wrapper3 = mount(
      {
        render (h) {
          return <Button><span>按钮</span></Button>
        },
      }
    )
    Vue.nextTick(() => {
      expect(wrapper3.find('.ant-btn').contains('.ant-btn-two-chinese-chars')).toBe(true)
      done()
    })
  })
  it('should change loading state instantly by default', () => {
    const DefaultButton = {
      data () {
        return {
          loading: false,
        }
      },
      methods: {
        enterLoading () {
          this.loading = true
        },
      },

      render () {
        return <Button loading={this.loading} onClick={this.enterLoading}>Button</Button>
      },
    }
    const wrapper = mount(DefaultButton)
    wrapper.trigger('click')
    Vue.nextTick(() => {
      expect(wrapper.findAll('.ant-btn-loading').length).toBe(1)
    })
  })

  it('should change loading state with delay', (done) => {
    const DefaultButton = {
      data () {
        return {
          loading: false,
        }
      },
      methods: {
        enterLoading () {
          this.loading = { delay: 1000 }
        },
      },

      render () {
        return <Button loading={this.loading} onClick={this.enterLoading}>Button</Button>
      },
    }
    const wrapper = mount(DefaultButton)
    wrapper.trigger('click')
    Vue.nextTick(() => {
      expect(wrapper.contains('.ant-btn-loading')).toBe(false)
      done()
    })
  })

  it('should support link button', () => {
    const wrapper = mount({
      render (h) {
        return <Button target='_blank' href='http://ant.design'>link button</Button>
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('fixbug renders {0} , 0 and {false}', () => {
    const wrapper = mount({
      render (h) {
        return <Button>{0}</Button>
      },
    })
    expect(wrapper.html()).toMatchSnapshot()

    const wrapper1 = mount({
      render (h) {
        return <Button>0</Button>
      },
    })
    expect(wrapper1.html()).toMatchSnapshot()

    const wrapper2 = mount({
      render (h) {
        return <Button>{false}</Button>
      },
    })
    expect(wrapper2.html()).toMatchSnapshot()
  })
})
