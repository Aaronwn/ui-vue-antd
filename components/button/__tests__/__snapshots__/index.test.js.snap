// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button fixbug renders {0} , 0 and {false} 1`] = `<button type="button" class="ant-btn ant-btn-default"><span>0</span></button>`;

exports[`Button fixbug renders {0} , 0 and {false} 2`] = `<button type="button" class="ant-btn ant-btn-default"><span>0</span></button>`;

exports[`Button fixbug renders {0} , 0 and {false} 3`] = `<button type="button" class="ant-btn ant-btn-default"></button>`;

exports[`Button renders Chinese characters correctly 1`] = `
<button type="button" class="ant-btn ant-btn-default">
  <i class="anticon anticon-search"></i><span>按钮</span></button>
`;

exports[`Button renders Chinese characters correctly 2`] = `
<button type="button" class="ant-btn ant-btn-default">
  <i class="anticon anticon-search"></i>按钮</button>
`;

exports[`Button renders correctly 1`] = `<button type="button" class="ant-btn ant-btn-default"><span>Follow</span></button>`;

exports[`Button should support link button 1`] = `<button target="_blank" href="http://ant.design" type="button" class="ant-btn ant-btn-default"><span>link button</span></button>`;
